test-utils/runtime-tester/README.md

```diff
@@ -0,0 +1,105 @@
+# Runtime test
+Runtime test is described by Scenario.
+Currently, scenario only supports one client, but you can specify the number of accounts through NetworcConfig.
+```rust
+#[derive(Serialize, Deserialize)]
+pub struct Scenario {
+    pub network_config: NetworkConfig,
+    pub blocks: Vec<BlockConfig>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct NetworkConfig {
+    pub seeds: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct BlockConfig {
+    pub height: BlockHeight,
+    pub transactions: Vec<TransactionConfig>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct TransactionConfig {
+    pub nonce: Nonce,
+    pub signer_id: AccountId,
+    pub receiver_id: AccountId,
+    pub signer: InMemorySigner,
+    pub actions: Vec<Action>,
+}
+```
+
+Scenario can be loaded from a json file or constructed in rust code.
+```rust
+pub fn from_file(path: &Path) -> io::Result<Scenario>
+```
+
+Scenario::run tries to produce all the described blocks and if succeeded returns RuntimeStats.
+```rust
+pub fn run(&self) -> Result<RuntimeStats, Error>
+```
+
+RuntimeStats contain stats for every produced block. Currently, only block production time is supported.
+```rust
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct RuntimeStats {
+    pub blocks_stats: Vec<BlockStats>,
+}
+
+#[derive(Serialize, Deserialize, Default, Debug)]
+pub struct BlockStats {
+    pub height: u64,
+    pub block_production_time: Duration,
+}
+```
+
+Be careful to remember, that block height should be positive and ascending.
+
+# Scenario Builder
+To easily create new scenarios in rust code use ScenarioBuilder.
+
+```rust
+impl ScenarioBuilder {
+    pub fn new(num_accounts: usize, use_in_memory_store: bool) -> Self
+
+    pub fn add_block(&mut self)
+
+    pub fn add_transaction(&mut self, signer_index: usize, receiver_index: usize, actions: Vec<Action>)
+
+    pub fn scenario(&self) -> &Scenario
+}
+```
+`new` -- Creates builder with an empty scenario with `num_accounts` accounts.
+`add_block` -- Adds empty block to the scenario with the next height (starting from 1).
+`add_transaction` -- Adds transaction to the last block in the scenario.
+`scenario` -- Returns a reference to the built scenario.
```

Comment:
These are API-level docs, they definitely should be specified as doc-comments in code:

```rust
impl ScenarioBuilder {
    /// Adds transaction to the last block in the scenario.
    pub fn add_transaction(
        &mut self,
        signer_index: usize,
        receiver_index: usize,
        actions: Vec<Action>,
    ) {
}
```

The reasoning is that docs and code gets desynchronized in no-time. As this is an internal tool, and we don't have a dedicated tech-writer to make sure it is up-to-date, it is almost guaranteed that this form of the docs becomes misleading very soon. 

The trick is to document as little as you can, and keep the docs as close to the thing they describe as possible.